### DISCUSSION TITLE
Handling WorkerLostError

### DISCUSSION THREAD

**Discussion by Santifire92** [celery/billiard#409] [General]
Hi, @auvipy !
I have a question, how can we handle this scenario where a worker gets lost abruptly like with a sigkill?
In the case of a timeout, i see you raise the exception so it gets propagated upwards, but with the worker lost, it just gets caught and passes as if nothing happened to the upper method caller.

Can this be raised/what can i do?
https://github.com/celery/billiard/blob/81cc94204ed70765a1f202b580e1063229af729c/billiard/pool.py#L1263-L1273

**Comment by thedrow** [comment]
When a worker is lost, a new one should be instantiated instead.
Is that not the case?

**Reply by Santifire92** [reply]
Hello, thanks so much for answering!
It DOES get reinstantiated, my problem is that it requeues the failed task in an infinite loop ignoring retries flag (i think because this exception doesnt get propagated) so if you have a task that always fails by lets say memory consumption, it will requeue and fail in an infinite loop.

How do you handle this scenario?

Task starts -> worker gets sigkilled -> task requeues
Task starts -> worker gets sigkilled -> task requeues
Task starts -> worker gets sigkilled -> task requeues
...

add infinitum

**Reply by thedrow** [reply]
This is the normal behavior for Billiard. You need to cancel the request to execute the task somehow.
If this is a Celery issue, revoking the task could do the trick.

**Reply by Santifire92** [reply]
Thanks so much for clarifying the behaviour!
